### DISCUSSION TITLE
Pandocomatic's future given pandoc's "defaults file" feature

### DISCUSSION THREAD

**Discussion by htdebeer** [htdebeer/pandocomatic#105] [Ideas]
By now, pandoc has the [defaults file](https://pandoc.org/MANUAL.html#defaults-files) feature. With a defaults file, you can put common pandoc options in an external file and re-use them. There's even inheritance! This is a great feature. Actually, the lack of this feature in earlier version of pandoc was one of the reasons to build pandocomatic! 

I think we should start using this defaults file feature, to keep pandocomatic aligned with the broader pandoc community. But what does this mean for pandocomatic? Should it continue as it is? Should it become sort of a makefile for pandoc and focus on the pre and post processing stages? Or something else? 

**Please share your ideas and wishes!**

On the top of my head, possible avenues to investigate:

- Remove the "pandoc" property of pandocomatic templates and use defaults files instead. Maybe the same for the "metadata" property.
- Put the pandocomatic templates in the defaults files somehow. For example, by adding the "preprocessor" and "postprocessor" properties, and "startup" and "cleanup", too.
- Make from pandocomatic a workflow-based document processing system with pandoc at its heart. 
- ...

**Comment by iandol** [comment]
I still find the more flexible inheritance and pre/post processing flexibility of pandocomatic much more useful than the defaults files (even though [I did make defaults of many of my pandocomatic recipes](https://github.com/iandol/dotpandoc/tree/master/defaults#why-i-still-prefer-pandocomatic), I never actually use them).

In addition, the ability to call a defaults file **from** the document metadata is missing ([and the issue is still mired without much movement](https://github.com/jgm/pandoc/issues/5870)). This makes the defaults system less flexible than pandocomatic, which can read the "recipe"[s] to use from the document file itself.

If it made pandocomatic easier to maintain, then using the defaults files for specifying the `pandoc` and `metadata` options while keeping the inheritance and pre/post-processing would be a feasible option, though it means dealing with several files where one used to suffice...
